### PR TITLE
Module5integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by derrick1451**
### Requirements
- In your PR description include the screenshots of your app's logs before and after the fix
- Before
Screenshot from 2023-09-18 17-50-47
Screenshot from 2023-09-18 17-51-13
- After
Screenshot from 2023-09-18 17-56-19
Screenshot from 2023-09-18 17-56-42
- make Integration specs tests on the following
  - User index page:
  - user show page:
  - User post index page:
  - Post show page:
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Higlights
- REsolved
- Hello reviewer am working on this project as a solo student hope this wont have any effect on your review
- 
Screenshot from 2023-09-19 15-52-45

### CODE REVIEW

File: spec/system/users/show.spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    user = User.create(Name: 'derrick', Photo: 'https://placehold.co/200x133', PostsCounter: 0,
+                       Bio: 'I am a software developer')
+    Post.create(Title: 'Post 1', Text: 'Post 1 content', author: user)
+    Post.create(Title: 'Post 2', Text: 'Post 2 content', author: user)
+    Post.create(Title: 'Post 3', Text: 'Post 3 content', author: user)
+
+    visit user_path(user)
+    click_link 'see all posts'
+    expect(page).to have_current_path(user_posts_path(user))
+  end
```
Comment (DammyShittu) [inline spec/system/users/show.spec.rb]: - [ ] Kindly add the tests for checking that when a user clicks on a post it takes them to the post show page.

- [ ] **[OPTIONAL]** Instead of repeating these data creations why not wrap the entire tests in a `describe` block, create the needed, and then use the `context` and `it` blocks to make your assertions?

  ```
  Rspec.describe 'User show page' do
    describe "GET #show" do
      user = ...
      post = ...

      visit_..._path
  
      context "When user visits so and so page" do
        it "displays the user's name" do
          ...
        end

        it "displays the profile picture" do
          ...
        end
      end
    end
  end

File: spec/system/posts/index.spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  describe 'links attached to the page' do
+    before { visit user_posts_path(@user) }
+
+    it 'redirects to the post show page when clicking on a post' do
+      post = @post1
+      click_link post.id
+      expect(page).to have_current_path(user_post_path(@user, post))
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/system/posts/index.spec.rb]: - [ ] Kindly add test for ensuring we can see the post title.

File: spec/system/posts/index.spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts Index Page', type: :system do
```
Comment (DammyShittu) [inline spec/system/posts/index.spec.rb]: - [ ] Your test examples do not run at all. You have the wrong file names. Your file names should be `_spec.rb`, not `.spec.rb`.

  `index.spec.rb` - ❌
  `index_spec.rb` - ✔️ 
  `show.spec.rb` - ❌
  `show_spec.rb` - ✔️ 

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @derrick1451,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -28,7 +30,7 @@
 
 
 
-
+
```
Comment (HAMMAS-SALEEM) [inline app/views/users/show.html.erb:36]: **[OPTIONA BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you fix the controller specs because i saw that some of them are failing 😨 Kindly fix them to make your controllers tested well 👍

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @derrick1451 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements
✔️ Great work on testing the edge cases 🔥
✔️ Great work on organizing your tests in dedicated files 👍

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
